backend/internal/storage/job_offers_db.go

```diff
@@ -6,6 +6,59 @@ import (
 	"gorm.io/gorm"
 )
 
+func (db DB) GetJobOffers(page string, limit string, jobtitle string, company string, city string, isRemote string) (v1beta.JobOffersResponse, error) {
+	offset, limitInt := Paginate(page, limit)
+	var nbHits int64
+
+	// Build the query
+	query := db.queryJobOffers().Order("jb.createdat DESC")
+
+	if jobtitle != "" {
+		query = query.Where("j.title LIKE ?", "%"+jobtitle+"%")
+	}
+
+	if company != "" {
+		query = query.Where("c.name LIKE ?", "%"+company+"%")
+	}
+
+	if city != "" {
+		query = query.Where("ct.name LIKE ?", "%"+city+"%")
+	}
```

Comment:
i feel like those if checks are painful for the app, if one of the element is "empty" the "LIKE %% " query statement is not going to break right ?

``` suggestion
	for key, item := range map[string]string{
		"j.title": jobtitle,
		"c.name":  company,
		"ct.name": city,
	} {
               query = query.Where("%s LIKE ?", "%"+item+"%")
        }
```